File: Solutions/php/y2015/solution01.php
<?php 

include dirname(__DIR__)."\Solution.php";

class Solution01 implements Solution {
    public function Part1($inputFile) {
        $result = 0;
        if ($file = fopen($inputFile, "r")) {
            while(!feof($file)) {
                $textperline = fgets($file);
                $chars = str_split($textperline);
                foreach ($chars as $char) {
                    $result += ($char === '(' ? 1 : -1);
                }    
            }
            fclose($file);
        }

        echo "Santa is on the ".$result."th floor.\n";
    }
    
    public function Part2($inputFile) {
        $result = 0;
        $counter = 0;

        if ($file = fopen($inputFile, "r")) {
            while(!feof($file)) {
                $textperline = fgets($file);
                $chars = str_split($textperline);
                foreach ($chars as $char) {
                    $result += ($char === '(' ? 1 : -1);
                    $counter += 1;

                    if($result < 0) {
                        break;
                    }
                }    
                // echo "Break while\n";
            }
            fclose($file);
        }

        echo "Santa found the basement after $counter tries\n";
    }
}
?>

File: Solutions/r/y2015/Solution01.r
part1 <- function(input_file) {
    data <- paste(readLines(input_file), collapse = "\n")
    result <- 0

    characters <- strsplit(data, split = "")
    for (i in characters[[1]]) {
        result <- result + if (i == "(") 1 else -1
    }

    print(paste("Santa is on the ", result, "th floor", sep = ""))
}

part2 <- function(input_file) {
    data <- paste(readLines(input_file), collapse = "\n")
    floor <- 0
    trycounter <- 0

    characters <- strsplit(data, split = "")
    for (i in characters[[1]]) {
        trycounter <- trycounter + 1
        floor <- floor + if (i == "(") 1 else -1
        if (floor < 0) {
            break
        }
    }

    print(paste("Santa found the basement after ", trycounter, " tries", sep = "")) # nolint
}